contracts/contracts/levels/DoubleEntryPoint.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+interface DelegateERC20 {
+  function delegateTransfer(address to, uint256 value, address origSender) external returns (bool);
+}
+
+interface IAgent {
+    function handleTransaction(address user) external returns(bool);
+}
+
+interface IForta {
+    function setAgent(address agentAddress) external;
+    function notify(address user) external returns(bool);
+    function raiseAlert(address user) external;
+}
+
+contract Forta is IForta {
+  mapping(address => IAgent) public usersAgent;
+  mapping(address => uint256) public agentRaisedAlerts;
+
+  function setAgent(address agentAddress) external override {
+      require(address(usersAgent[msg.sender]) == address(0), "Agent already set");
+      usersAgent[msg.sender] = IAgent(agentAddress);
+  }
+
+  function notify(address user) external override returns(bool) {
```

Comment:
I think this function should never revert, otherwise it opens up the possibility of solving the level in some other way (I think?). If there is no agent, it should simply return. The `handleTransaction` should be executed in `try catch`.